src/main/java/net/javadiscord/javabot/events/MessageLinkListener.java

```diff
@@ -0,0 +1,63 @@
+package net.javadiscord.javabot.events;
+
+import net.dv8tion.jda.api.EmbedBuilder;
+import net.dv8tion.jda.api.JDA;
+import net.dv8tion.jda.api.entities.Guild;
+import net.dv8tion.jda.api.entities.GuildChannel;
+import net.dv8tion.jda.api.entities.Message;
+import net.dv8tion.jda.api.entities.MessageEmbed;
+import net.dv8tion.jda.api.events.message.MessageReceivedEvent;
+import net.dv8tion.jda.api.hooks.ListenerAdapter;
+import net.dv8tion.jda.api.requests.RestAction;
+import net.javadiscord.javabot.Bot;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Arrays;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class MessageLinkListener extends ListenerAdapter {
+
+	private final Pattern MESSAGE_URL_PATTERN = Pattern.compile("https://((?:canary|ptb)\\.)discord.com/channels/[0-9]+/[0-9]+/[0-9]+");
+
+	@Override
+	public void onMessageReceived(@NotNull MessageReceivedEvent event) {
+		if (event.getAuthor().isBot() || event.getAuthor().isSystem()) return;
+		Matcher matcher = MESSAGE_URL_PATTERN.matcher(event.getMessage().getContentRaw());
+		if (matcher.find()) {
+			var optional = parseMessageUrl(matcher.group(), event.getJDA());
+			optional.ifPresent(action -> action.queue(m -> event.getMessage().replyEmbeds(buildUrlEmbed(m)).queue()));
+		}
+	}
+
+	private MessageEmbed buildUrlEmbed(Message m) {
+		return new EmbedBuilder()
+				.setAuthor("Jump to Original", m.getJumpUrl())
+				.setColor(Bot.config.get(m.getGuild()).getSlashCommand().getDefaultColor())
+				.setDescription(m.getContentRaw())
+				.setTimestamp(m.getTimeCreated())
+				.setFooter(m.getAuthor().getAsTag(), m.getAuthor().getEffectiveAvatarUrl())
+				.build();
+	}
+
+	private Optional<RestAction<Message>> parseMessageUrl(String url, JDA jda) {
+		RestAction<Message> optional = null;
+		try {
+			var arr = url.split("/");
+			String[] segments = Arrays.copyOfRange(arr, 4, arr.length);
+			if (jda.getGuilds().stream().map(Guild::getId).toList().contains(segments[0])) {
```

Comment:
maybe only check the guild the message is originating from or guilds the message author is in.

Reply:
Yeah, but that's what it already does? The link consists of `.../channels/{guildId}/{channelId}/{messageId}`. Its just checking if the bot itself is in that particular guild and could, theoretically, retrieve the message

Reply:
This just checks if the bot is in the guild, not if the guild matches with the message from the event or the user sending the link.

Reply:
yea, but is that necessary? As long as the Bot can retrieve the message. This way, it would theoretically even work across multiple servers, which is not bad imo.